### PR TITLE
[issue: #651] Adding initialization for non requiered containers in add/put method

### PR DESCRIPTION
**Pull request by boyi01**
Many thanks for submitting your Pull Request :heart:!

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.md)
- [x] Your code is properly formatted according to [our code style](CONTRIBUTING.md#code-style)
- [x] Pull Request title contains the target branch if not targeting main: `[0.9.x] Subject`
- [x] Pull Request contains link to the issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

When in the model there is a field, that is for example a list and is not diffined as requierd in the yaml, then in the add method there is no initialization, if this field is null.

Fix it as openapi does it, in each "add" method for a container, generate a check if the container is null, if it's null instantiate a new empty container. 

@ricardozanini 
Is there an option to backport this to 2.2.16? Or should we do this local in our Nexus?

### CODE REVIEW

**Comment by ricardozanini** [comment]
@boyi01 thank you! 

> Is there an option to backport this to 2.2.16? Or should we do this local in our Nexus?

No, we are likely to release a new `2.4.0` version with the latest bumps and PRs.

**Review by ricardozanini — APPROVED** [review]

**Comment by ricardozanini** [comment]
@fjtirado I remember you fixed some arrays problem in the past. Is this related?

**Review by mcruzdev — APPROVED** [review]
LGTM! Thank @boyi01

**Comment by boyi01** [comment]
@ricardozanini  Oky then we have have release our own Version für Quarkus 3.2 with Java11.

Thank you for the fast responding and the work on this repository.

File: integration-tests/array-enum/src/test/java/io/quarkiverse/openapi/generator/it/ArrayEnumTest.java [hunk 1]
```diff
@@ -21,4 +22,18 @@ class ArrayEnumTest {
     void apiIsBeingGenerated() {
         assertThat(api).isNotNull();
     }
+
+    @Test
+    void modelNonRequieredFieldTest() {
```
Comment (hbelmiro) [inline integration-tests/array-enum/src/test/java/io/quarkiverse/openapi/generator/it/ArrayEnumTest.java:27]:
```suggestion
    void modelNonRequiredFieldTest() {
```

**Comment by hbelmiro** [comment]
@all-contributors add @boyi01 for code and test.

**Pull request merged**